public/API/API_GetUserSetRequests.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ *  API_GetUserSetRequests - gets user's set request list
+ *    u : username
+ *
+ *  array      RequestedSets
+ *    object     [value]
+ *      int        GameID                     unique identifier of the game
+ *      string     Title                      set title
+ *      int        ConsoleID                  the id of the console that the set is for
+ *      string     ConsoleName                the name of the console that the set is for
+ *      string     ImageIcon                  the url of the sets icon
+ *  int        TotalRequests              maximum number of requests a user has
+ *  int        PointsForNext              number of points remaining until maximum request increase
+ */
+
+use App\Models\Game;
+use App\Models\User;
+use App\Models\UserGameListEntry;
+
+$user = User::firstWhere('User', request()->query('u'));
+if (!$user) {
+    return response()->json([]);
+}
+
+$requestedSets = UserGameListEntry::where('user_id', $user->ID)
+    ->where('type', 'achievement_set_request')
+    ->pluck('GameID')
+    ->toArray();
+
+$games = Game::with('system')
+    ->whereIn('ID', $requestedSets)
+    ->get();
```

Comment:
This is significantly better than before, but can be optimized even further.

If you combine these two queries into a single query, three benefits are unlocked:
* Memory use is reduced because we can remove `->toArray()`.
* We can avoid a large `WHERE IN` lookup.
* We only need to rely on a single composite index.

Reply:
I have condensed this down to a single query and combined it with the map for the final response.

Reply:
Much better. Can be optimized even further though!

```php
// no need for map here, the fields are already aliased

// ->get()
// ->toArray();
// ^^^ faster
->map(function ($gameData) {
    return [
        'GameID' => $gameData->GameID,
        'Title' => $gameData->Title,
        'ConsoleID' => $gameData->ConsoleID,
        'ConsoleName' => $gameData->ConsoleName,
        'ImageIcon' => $gameData->ImageIcon,
    ];
});
```

```php
->orderBy('GameData.sort_title', 'asc') // nit: 'asc' is the default
```

Reply:
Map has been removed along with explicit 'asc'. I thought I would include it for readability but have removed it based on the nit.